### PR TITLE
fix: Error navigation blockers

### LINKED ISSUE #22283: [Bug]: The debugger opens up when we perform upload from FilePicker overlay
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The debugger opens up when we perform upload from FilePicker overlay

### Steps To Reproduce

1. DnD file picker
2. Add onFileSelected action
3. Click on filepicker , upload some file , and click on upload
the debugger opens up

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Release

### PR DESCRIPTION
**Pull request by ChandanBalajiBP**
## Description

> Remove debugger from preview mode
> Remove debugger in welcome tour
> Don't open debugger on `onpageload` action.

Fixes #22283 
#22281 
#22275 

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

- Manual
- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/pages/Editor/APIEditor/CommonEditorForm.tsx [hunk 1]
```diff
@@ -697,10 +697,15 @@ function CommonEditorForm(props: CommonFormPropsWithExtraParams) {
...
-  const showDebugger = useSelector(
+  const debuggerState = useSelector(
     (state: AppState) => state.ui.debugger.isOpen,
   );
+  const isPreviewMode = useSelector(
+    (state: AppState) => state.ui.editor.isPreviewMode,
+  );
+
+  // Debugger render flag
+  const showDebugger = debuggerState && !isPreviewMode;
```
[the remark this answers is not in the document]
Comment (ChandanBalajiBP) [reply app/client/src/pages/Editor/APIEditor/CommonEditorForm.tsx:708]: Right, will update this.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug22281_WelcomeTour_spec.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { ObjectsRegistry } from "../../../../support/Objects/Registry";
+
+const locator = ObjectsRegistry.CommonLocators,
+  agHelper = ObjectsRegistry.AggregateHelper,
+  debuggerHelper = ObjectsRegistry.DebuggerHelper;
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug22281_WelcomeTour_spec.ts:5]: Use imports from ObjectsCore & reduce the objects declaration in spec files.

File: app/client/cypress/integration/Regression_TestSuite/ServerSideTests/OnLoadTests/OnLoadActions_Spec.ts [hunk 3]
```diff
@@ -167,6 +167,8 @@ describe("Layout OnLoad Actions tests", function () {
     });
 
     deployMode.NavigateBacktoEditor();
+    //Verify if debugger is closed after failure of onpageload actions.
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ServerSideTests/OnLoadTests/OnLoadActions_Spec.ts:170]: Add bug id to comment.

File: app/client/cypress/support/Objects/CommonLocators.ts [hunk 4]
```diff
@@ -178,4 +178,10 @@ export class CommonLocators {
   _editorBackButton = ".t--close-editor";
   _evaluateMsg = ".t--evaluatedPopup-error";
   _canvas = "[data-testid=widgets-editor]";
+  _enterPreviewMode = "[data-cy='edit-mode']";
+  _exitPreviewMode = "[data-cy='preview-mode']";
+  _appsmithLogo = ".t--appsmith-logo";
+  _createNewApplicationButton = ".t--new-button";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Objects/CommonLocators.ts:184]: Repetitive locators. Can be removed. `homePage._homePageAppCreateBtn` can be used.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug22281_WelcomeTour_spec.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    //Enter preview mode
+    agHelper.GetNClick(locator._enterPreviewMode);
+    //verify debugger is not present
+    agHelper.AssertElementAbsence(locator._errorTab);
+    //Exit preview mode
+    agHelper.GetNClick(locator._exitPreviewMode);
+    //verify debugger is present
+    agHelper.GetNAssertContains(locator._errorTab, "Errors");
+  });
+  it("Bug: 22282: Debugger should not open by default in welcome tour", function () {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug22281_WelcomeTour_spec.ts:20]: This is `22281` right?

File: app/client/cypress/integration/Regression_TestSuite/ServerSideTests/OnLoadTests/OnLoadActions_Spec.ts (hunk 3 above)
Comment (ChandanBalajiBP) [reply app/client/cypress/integration/Regression_TestSuite/ServerSideTests/OnLoadTests/OnLoadActions_Spec.ts]: Updated

File: app/client/cypress/support/Objects/CommonLocators.ts (hunk 4 above)
Comment (ChandanBalajiBP) [reply app/client/cypress/support/Objects/CommonLocators.ts]: Updated

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug22281_WelcomeTour_spec.ts (hunk 5 above)
Comment (ChandanBalajiBP) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug22281_WelcomeTour_spec.ts]: Updated

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug22281_WelcomeTour_spec.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    //verify debugger is not present
+    _.agHelper.AssertElementAbsence(_.locators._errorTab);
+    //Exit preview mode
+    _.agHelper.GetNClick(_.locators._exitPreviewMode);
+    //verify debugger is present
+    _.agHelper.GetNAssertContains(_.locators._errorTab, "Errors");
+  });
+  it("Bug: 22281: Debugger should not open by default in welcome tour", function () {
+    //Get back to application page
+    _.agHelper.GetNClick(_.locators._appsmithLogo);
```
[the remark this answers is not in the document]
Comment (ChandanBalajiBP) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug22281_WelcomeTour_spec.ts]: Updated

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress looks good.

**Review by Aishwarya-U-R — APPROVED** [review]
Approving Cypress.

**Pull request merged**